scripts/un/sdg/process.py

```diff
@@ -0,0 +1,292 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''Generates mcf and csv/tmcf for sdg-dataset data.
+
+Produces:
+* schema/ folder: 
+- measurement_method.mcf
+- schema.mcf (classes and enums)
+- sdg.textproto (vertical spec)
+- series.mcf (series mcf)
+- sv.mcf
+- unit.mcf
+* csv/ folder: 
+- [CODE].csv
+
+Usage: python3 process.py
+'''
+import os
+import pandas as pd
+import sys
+
+sys.path.append(
+    os.path.dirname(os.path.dirname(os.path.dirname(
+        os.path.abspath(__file__)))))
+from un.sdg import util
+
+
+def get_geography(code, type):
+    '''Returns dcid of geography.
+
+    Args:
+        code: Geography code.
+        type: Geography type.
+
+    Returns:
+        Geography dcid.
+    '''
+
+    # Currently only support Country and City.
+    if type == 'Country' and code in util.PLACES:
+        return 'dcs:country/' + util.PLACES[code]
+    elif type == 'City' and code in util.CITIES and util.CITIES[code]:
+        return 'dcs:' + util.CITIES[code]
+    return ''
+
+
+def get_unit(units, base_period):
+    '''Returns dcid of unit.
+
+    Args:
+        unit: Unit.
+        base_period: Base period of unit.
+
+    Returns:
+        Unit dcid.
+    '''
+    if util.is_valid(base_period):
+        return f'[{units} {base_period}]'
+    return 'dcs:SDG_' + units
+
+
+def get_measurement_method(row):
+    '''Returns dcid of measurement method.
+
+    Args:
+        row: Input DataFrame row.
+    
+    Returns:
+        Measurement method dcid.
+    '''
+    mmethod = ''
+    if util.is_valid(row['NATURE']):
+        mmethod += '_' + str(row['NATURE'])
+    if util.is_valid(row['OBSERVATION_STATUS']):
+        mmethod += '_' + str(row['OBSERVATION_STATUS'])
+    if util.is_valid(row['REPORTING_TYPE']):
+        mmethod += '_' + str(row['REPORTING_TYPE'])
+    return 'SDG' + mmethod
+
+
+def process(input_dir, schema_dir, csv_dir):
+    '''Generates mcf, csv/tmcf artifacts.
+    
+    Args:
+        input_dir: Path to input xlsx files.
+        schema_dir: Path to output schema files.
+        csv_dir: Path to output csv files.
+    '''
+    with open(os.path.join(schema_dir, 'series.mcf'), 'w') as f_series:
```

Comment:
Any chance you can factor this verrrry long function into helpers for better readability? :)

Reply:
added more comments for this and can refactor as follow up!